### ISSUE TITLE
Request: Formatter to allow different configuration

### ISSUE THREAD

**Issue by alansouzati** [eslint/eslint#7796]
Currently all the Formatters don't allow configuration object to be passed.

https://github.com/eslint/eslint/blob/master/lib/formatters/stylish.js#L28

In my case I would like to add a flag to print or not the total for the test results.

**Tell us about your environment**

* **ESLint Version:** v3.12.2
* **Node Version:**  v6.9.1
* **npm Version:**  v3.10.9

**What parser (default, Babel-ESLint, etc.) are you using?** Default

Are you guys open to adding options capabilities to Formatter? I would be happy to contribute this as a pull request.

```javascript

formatter(report.results, {
  printTotal: false
})

```

To remove the total from the console output, I'm currently creating my own formatter based on the great stylish formatter available in eslint.

**Comment by platinumazure** [comment]
I'd love to see formatter option support be added to the CLI, but I think the general team preference on this is to create and use custom formatters.

**Comment by alansouzati** [comment]
Thanks for chiming in @platinumazure. Just to provide a little bit more context, I'm building an eslint lib that will execute things in parallel, the reason I want to remove the total is because each of my child process will print the total, where the behavior I want is to print total only once when all my promises have been resolved.

I'm definitely see the point of avoiding too many options and configurations. If the guideline is to create my own formatter I'm fine with that. I just thought to raise a request here and see what the community thinks.

**Comment by platinumazure** [comment]
If you're looking to use output in a Node program that does more steps after running ESLint, you would be better served using our CLIEngine API directly or using a formatter like JSON or XML which is more machine-readable. As far as your console output goes, you can certainly base that off our `stylish` formatter if you like.

**Comment by alansouzati** [comment]
I'm definitely using CLIEngine API. I already have the results as a JSON object. 

I'm currently looking for a built-in solution that would allow to print stylish results without adding total to the output.

**Comment by vitorbal** [comment]
@platinumazure is right, I think your solution of using a custom formatter is what the team would recommend in this case. Plus, I don't really see the point in adding customization for formatters in the core just to show the total of errors or not. Sounds like a very specific use case.

Having said that, I'd be interested in your ESLint lib to run things in parallel, sounds like a nice idea!

**Comment by alansouzati** [comment]
Cool. I will let you know when I publish it

**Comment by alansouzati** [comment]
@vitorbal there you go https://github.com/alansouzati/eslint-parallel

Any feedback would be appreciated

**Comment by platinumazure** [comment]
I'm going to close this issue since the original question has been answered, but please feel free to continue the discussion as needed.
